File: apps/backend/src/services/email.js
import { Resend } from 'resend'

const resend = process.env.RESEND_API_KEY ? new Resend(process.env.RESEND_API_KEY) : null;
const DEVELOPER_EMAIL = process.env.DEVELOPER_EMAIL || 'developer@example.com'

export async function sendCriticalFeedbackEmail(feedback) {
  try {
    if (!resend) return;
    await resend.emails.send({
      from: 'Pegasus Feedback <feedback@pegasus.dev>',
      to: DEVELOPER_EMAIL,
      subject: `🚨 Critical Feedback: ${feedback.feature_category}`,
      html: `
        <h2>Critical Feedback Received</h2>
        <p><strong>Priority:</strong> ${feedback.priority.toUpperCase()}</p>
        <p><strong>Feature:</strong> ${feedback.feature_category}${feedback.custom_feature ? ` - ${feedback.custom_feature}` : ''}</p>
        <p><strong>Issue Type:</strong> ${feedback.issue_type}</p>
        <p><strong>User Email:</strong> ${feedback.user_email || 'Not provided'}</p>
        <p><strong>Marked as Urgent:</strong> ${feedback.is_urgent ? 'Yes' : 'No'}</p>
        <hr>
        <h3>Description:</h3>
        <p>${feedback.description}</p>
        <hr>
        <p><small>Browser: ${feedback.browser_info || 'Unknown'}</small></p>
        <p><small>Submitted: ${new Date(feedback.created_at).toLocaleString()}</small></p>
      `
    })
  } catch (error) {
    console.error('Failed to send critical feedback email:', error)
  }
}

export async function sendWeeklyDigest(feedbackList) {
  if (feedbackList.length === 0) return

  const criticalFeedback = feedbackList.filter(f => f.priority === 'critical')
  const normalFeedback = feedbackList.filter(f => f.priority === 'normal')

  const groupByCategory = (items) => {
    return items.reduce((acc, item) => {
      const cat = item.feature_category
      if (!acc[cat]) acc[cat] = []
      acc[cat].push(item)
      return acc
    }, {})
  }

  const criticalByCategory = groupByCategory(criticalFeedback)
  const normalByCategory = groupByCategory(normalFeedback)

  const renderFeedbackSection = (title, grouped) => {
    if (Object.keys(grouped).length === 0) return ''

    return `
      <h3>${title}</h3>
      ${Object.entries(grouped).map(([category, items]) => `
        <h4>${category} (${items.length})</h4>
        <ul>
          ${items.map(item => `
            <li>
              <strong>${item.issue_type}</strong>: ${item.description.substring(0, 100)}${item.description.length > 100 ? '...' : ''}
              ${item.user_email ? `<br><small>From: ${item.user_email}</small>` : ''}
            </li>
          `).join('')}
        </ul>
      `).join('')}
    `
  }

  try {
    if (!resend) return;
    await resend.emails.send({
      from: 'Pegasus Feedback <feedback@pegasus.dev>',
      to: DEVELOPER_EMAIL,
      subject: `📊 Weekly Feedback Digest - ${feedbackList.length} items`,
      html: `
        <h2>Weekly Feedback Digest</h2>
        <p>Summary of feedback from the past week</p>
        <p><strong>Total Feedback:</strong> ${feedbackList.length}</p>
        <p><strong>Critical:</strong> ${criticalFeedback.length} | <strong>Normal:</strong> ${normalFeedback.length}</p>
        <hr>
        ${renderFeedbackSection('🚨 Critical Issues', criticalByCategory)}
        ${renderFeedbackSection('📝 Normal Feedback', normalByCategory)}
        <hr>
        <p><small>Generated: ${new Date().toLocaleString()}</small></p>
      `
    })
  } catch (error) {
    console.error('Failed to send weekly digest:', error)
  }
}

File: apps/python-intelligence/app/services/email.py
from __future__ import annotations

from app.config import settings


async def send_email(to: str, subject: str, html: str) -> bool:
    """Send email via Resend."""
    if not settings.resend_api_key:
        print(f"[Email] No RESEND_API_KEY configured, skipping email to {to}")
        return False

    import httpx

    try:
        async with httpx.AsyncClient() as client:
            resp = await client.post(
                "https://api.resend.com/emails",
                headers={
                    "Authorization": f"Bearer {settings.resend_api_key}",
                    "Content-Type": "application/json",
                },
                json={
                    "from": "Pegasus <notifications@pegasus.dev>",
                    "to": [to],
                    "subject": subject,
                    "html": html,
                },
            )
            return resp.is_success
    except Exception as e:
        print(f"[Email] Failed to send: {e}")
        return False


async def send_critical_feedback(feedback: dict) -> None:
    """Send critical feedback notification to the developer."""
    await send_email(
        to=settings.developer_email,
        subject=f"🚨 Critical Feedback: {feedback.get('feature_category', 'Unknown')}",
        html=f"""
        <h2>Critical Feedback Received</h2>
        <p><strong>Priority:</strong> {feedback.get('priority', 'normal').upper()}</p>
        <p><strong>Feature:</strong> {feedback.get('feature_category', '')}</p>
        <p><strong>User Email:</strong> {feedback.get('user_email', 'Not provided')}</p>
        <hr>
        <p>{feedback.get('description', '')}</p>
        """,
    )


async def send_weekly_digest(feedback_list: list[dict]) -> None:
    """Send weekly feedback digest."""
    if not feedback_list:
        return

    critical = [f for f in feedback_list if f.get("priority") == "critical"]
    normal = [f for f in feedback_list if f.get("priority") == "normal"]

    html = f"""
    <h2>Weekly Feedback Digest</h2>
    <p><strong>Total:</strong> {len(feedback_list)} | <strong>Critical:</strong> {len(critical)} | <strong>Normal:</strong> {len(normal)}</p>
    """
    if critical:
        html += "<h3>🚨 Critical Issues</h3><ul>"
        for item in critical:
            html += f"<li><strong>{item.get('issue_type')}</strong>: {item.get('description', '')[:200]}</li>"
        html += "</ul>"

    await send_email(
        to=settings.developer_email,
        subject=f"📊 Weekly Feedback Digest - {len(feedback_list)} items",
        html=html,
    )
